src/http/server.ts

```diff
@@ -42,26 +42,29 @@ async function runServices(
   request: Drash.Request,
   response: Drash.Response,
   serviceMethod: "runBeforeResource" | "runAfterResource",
-): Promise<{ err: Error | null; end: boolean }> {
-  let err: Error | null = null;
+): Promise<boolean> {
   let end = false;
+
+  // There are two ways a service can short-circuit the
+  // request-resource-response lifecycle:
+  //
+  //   1. The service throws an error.
+  //   2. The service calls `this.end()`.
+  //
+  // If the service throws an error, then the request handler we pass in to `new
+  // StdServer()` will catch it and return a response.
+  //
+  // If the service calls `this.send()`, then the request handler we pass in to
+  // `new StdServer()` will return `new Response()`.
```

Comment:
good shout, now we dont need to hold onto the error until other services are ran, it can be caught by our main lifecycle logic